src/main/java/me/makkuusen/timing/system/api/TimingSystemAPI.java

```diff
@@ -61,4 +64,78 @@ public static TPlayer getTPlayer(UUID uuid) {
     public static void teleportPlayerAndSpawnBoat(Player player, Track track, Location location){
         ApiUtilities.teleportPlayerAndSpawnBoat(player, track, location);
     }
+
+    public static List<Heat> getHeats(){
+        return EventDatabase.getHeats();
+    }
+
+    public static List<Heat> getRunningHeats() {
+        return EventDatabase.getHeats().stream().filter(Heat::isActive).collect(Collectors.toList());
+    }
+
+    public static List<DriverDetails> getAllDriverDetailsFromHeat(Heat heat) {
+
+        List<DriverDetails> driverDetails = new ArrayList<>();
+        Driver first = null;
+        Driver previous = null;
+        for (Driver d : heat.getLivePositions()) {
+            driverDetails.add(getDriverDetailsFromHeatWithTimes(heat, d.getTPlayer().getUniqueId(), first, previous));
+            if (first == null) {
+                first = d;
+            }
+            previous = d;
+        }
+
+        return driverDetails;
+    }
+
+    private static DriverDetails getDriverDetailsFromHeatWithTimes(Heat heat, UUID driverUuid, Driver previousDriverCompare, Driver leaderDriverCompare) {
+        var driverDetails = getDriverDetailsFromHeat(heat, driverUuid, previousDriverCompare.getTPlayer().getUniqueId());
+        var driver = heat.getDrivers().get(driverUuid);
+
+        if (leaderDriverCompare != null) {
+            driverDetails.setGapFromLeader(driver.getTimeGap(leaderDriverCompare));
+        }
+
+        return driverDetails;
+    }
+
+    public static DriverDetails getDriverDetailsFromHeat(Heat heat, UUID driverUuid, UUID driverCompare) {
+        var driver = heat.getDrivers().get(driverUuid);
+        var driverDetails = new DriverDetails();
+
+        driverDetails.setName(driver.getTPlayer().getName());
+        driverDetails.setUuid(driverUuid.toString());
+        driverDetails.setTeamColor(driver.getTPlayer().getHexColor());
+        driverDetails.setOffline(driver.getTPlayer().getPlayer() == null);
+        if (!driverDetails.isOffline()) {
+            driverDetails.setInpit(driver.isInPit(driver.getTPlayer().getPlayer().getLocation()));
+        }
+        driverDetails.setLaps(driver.getLaps().size());
+        driverDetails.setPits(driver.getPits());
+        driverDetails.setPosition(driver.getPosition());
+        driverDetails.setStartPosition(driver.getStartPosition());
+
+        if (driverCompare != null) {
+            var previousDriverCompare = heat.getDrivers().get(driverCompare);
+            driverDetails.setGap(driver.getTimeGap(previousDriverCompare));
+        }
+
+        return driverDetails;
+    }
+
+    public static DriverDetails getDriverDetailsOfClosestPlayerFromHeat(Heat heat, UUID playerToCheckClosestTo) {
+        Player player = Database.getPlayer(playerToCheckClosestTo).getPlayer();
+        if (player == null) {
+            return new DriverDetails();
+        }
+        var driver = EventDatabase.getClosestDriverForSpectator(player);
```

Comment:
Looks like you already designed this as my comment explained on the pull request